### PR TITLE
Add isRequired and isNotRequired assertions

### PR DESCRIPTION
**Pull request by scalvert**
Implements #56 

Fills in the API for
- `isRequired`
- `isNotRequired`

TODO:
- [x] implement `isRequired`
- [x] implement `isNotRequired`
- [x] write tests
- [x] write docs
- [x] regenerate docs [rendered](https://github.com/scalvert/qunit-dom/blob/3470f2e776206017ad6fb080256a20a43147ecae/API.md)

### CODE REVIEW

File: lib/assertions/is-not-required.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import elementToString from '../helpers/element-to-string';
+
+export default function notRequired(message) {
+  let element = this.findTargetElement();
+  if (!element) return;
+
+  let result = element.required === false;
+  let actual = element.required === true ? 'required' : 'not required';
```
Comment (Turbo87) [inline lib/assertions/is-not-required.js]: I think you can just reuse the `result` variable here instead of comparing the property again

Comment (scalvert) [reply lib/assertions/is-not-required.js]: Done. Also updated `is-required.js` to do the same. FWIW, this is what `is-checked` does, so we may want to update that too separately.

Comment (Turbo87) [reply lib/assertions/is-not-required.js]: yeah, agreed. I merged it and planned to do a small cleanup round later :)

File: lib/assertions/is-not-required.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import elementToString from '../helpers/element-to-string';
+
+export default function notRequired(message) {
+  let element = this.findTargetElement();
+  if (!element) return;
+
+  let result = element.required === false;
```
Comment (Turbo87) [inline lib/assertions/is-not-required.js]: what happens if the element is not an `<input>` element? would `required` be `undefined` then and fail this check?

Comment (scalvert) [reply lib/assertions/is-not-required.js]: Good catch. I've added a conditional, which will throw if an element is passed that is not one of `HTMLInputElement`, `HTMLTextAreaElement`, or `HTMLSelectElement`.

Comment (scalvert) [reply lib/assertions/is-not-required.js]: And a test...

**Comment by scalvert** [comment]
@Turbo87 anything else I need to do to push this in?

**Comment by Turbo87** [comment]
> anything else I need to do to push this in?

remind me about it 😜

**Pull request merged**